src/test/java/catan/controllers/testcases/play/UseCardYearOfPlentyTest.java

```diff
@@ -0,0 +1,326 @@
+
+import static catan.domain.model.dashboard.types.HexType.BRICK;
+import static catan.domain.model.dashboard.types.HexType.EMPTY;
+import static catan.domain.model.dashboard.types.HexType.SHEEP;
+import static catan.domain.model.dashboard.types.HexType.STONE;
+import static catan.domain.model.dashboard.types.HexType.WHEAT;
+import static catan.domain.model.dashboard.types.HexType.WOOD;
+import static catan.domain.model.game.types.DevelopmentCard.YEAR_OF_PLENTY;
+import static java.util.Arrays.asList;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+
+//@SpringApplicationConfiguration(classes = {TestApplicationConfig.class, RequestResponseLogger.class})  // if needed initial request and JSON response logging:
+//@SpringApplicationConfiguration(classes = TestApplicationConfig.class)
+@SpringApplicationConfiguration(classes = TestApplicationConfig.class)
+@WebIntegrationTest("server.port:8091")
+public class UseCardYearOfPlentyTest extends PlayTestUtil {
+
+    public static final String USER_NAME_1 = "user1_UseCardYearOfPlentyTest";
+    public static final String USER_PASSWORD_1 = "password1";
+    public static final String USER_NAME_2 = "user2_UseCardYearOfPlentyTest";
+    public static final String USER_PASSWORD_2 = "password2";
+    public static final String USER_NAME_3 = "user3_UseCardYearOfPlentyTest";
+    public static final String USER_PASSWORD_3 = "password3";
+
+    private static boolean initialized = false;
+
+    @Autowired
+    private RandomUtil randomUtil;
+
+    private Scenario scenario;
+
+    @Before
+    public void setup() {
+        if (!initialized) {
+            registerUser(USER_NAME_1, USER_PASSWORD_1);
+            registerUser(USER_NAME_2, USER_PASSWORD_2);
+            registerUser(USER_NAME_3, USER_PASSWORD_3);
+            initialized = true;
+        }
+
+        scenario = new Scenario((RandomUtilMock) randomUtil);
+    }
+
+    @Test
+    public void should_successfully_give_two_different_resources() {
+        playPreparationStageAndBuyCardYearOfPlentyAndPassCycle()
+                .startTrackResourcesQuantity()
+                .USE_CARD_YEAR_OF_PLENTY(1, "BRICK", "STONE").successfully()
+                .getGameDetails(1).gameUser(1).resourcesQuantityChangedBy(1, 0, 0, 0, 1);
+    }
+
+    @Test
+    public void should_successfully_give_two_same_resources() {
+        playPreparationStageAndBuyCardYearOfPlentyAndPassCycle()
+                .startTrackResourcesQuantity()
+                .USE_CARD_YEAR_OF_PLENTY(1, "WHEAT", "WHEAT").successfully()
+                .getGameDetails(1).gameUser(1).resourcesQuantityChangedBy(0, 0, 0, 2, 0);
+    }
+
+    @Test
+    public void should_successfully_use_the_card_if_player_bought_one_in_the_previous_move_and_one_in_the_current() {
+        playPreparationStageAndBuyCardYearOfPlentyAndPassCycle()
+                .nextRandomDevelopmentCards(asList(YEAR_OF_PLENTY))
+                .BUY_CARD(1)
```

Comment:
test case works good but it is not clear where did I have got the resources for second card. how about splitting a method playPreparationStageAndBuyCardYearOfPlentyAndPassCycle() on several parts?

for example:
playPreparationStage()
   .nextRandomDiceValues(asList(1, 1))   // P1, P2, P3: --
   .THROW_DICE(1)
   .giveResourcesForDevCardToFirstPlayerAndBuyIt(YEAR_OF_PLENTY)
   .giveResourcesForDevCardToFirstPlayerAndBuyIt(YEAR_OF_PLENTY)

also create separate method passCycleAfterFirstPlayerTrewDice() when needed.